cosmos/config.py

```diff
@@ -196,24 +219,21 @@ def validate_project(self) -> None:
         if self.dbt_project_path:
             project_yml_path = self.dbt_project_path / "dbt_project.yml"
             mandatory_paths = {
```

Comment:
The first time I saw this, I was confused about why we're doing things this way. It may be worth to add some comment on why